pkg/crypto/tinkcrypto/crypto.go

```diff
@@ -221,34 +226,34 @@ func (t *Crypto) WrapKey(cek, apu, apv []byte, recPubKey *cryptoapi.PublicKey,
 		opt(pOpts)
 	}
 
-	c, err := t.kw.getCurve(recPubKey.Curve)
+	wk, err := t.deriveKEKAndWrap(cek, apu, apv, pOpts.SenderKey(), recPubKey, pOpts.UseXC20PKW())
 	if err != nil {
-		return nil, fmt.Errorf("wrapKey: failed to get curve of recipient key: %w", err)
+		return nil, fmt.Errorf("wrapKey: %w", err)
 	}
 
-	pubKey := &ecdsa.PublicKey{
-		Curve: c,
-		X:     new(big.Int).SetBytes(recPubKey.X),
-		Y:     new(big.Int).SetBytes(recPubKey.Y),
-	}
-
-	ephemeralPriv, err := t.kw.generateKey(pubKey.Curve)
-	if err != nil {
-		return nil, fmt.Errorf("wrapKey: failed to generate EPK: %w", err)
-	}
-
-	return t.deriveKEKAndWrap(cek, apu, apv, pOpts.SenderKey(), ephemeralPriv, pubKey, recPubKey.KID)
+	return wk, nil
 }
 
 // UnwrapKey unwraps a key in recWK using ECDH (ES or 1PU) with recipient private key kh.
-// The optional 'wrapKeyOpts' specifies the sender kh for 1PU key unwrapping.
-// Note, if the option was used in WrapKey(), then it must be set here as well for a successful unwrapping.
 // This function is used with the following parameters:
-//  - Key Unwrapping: ECDH-ES (no options)/ECDH-1PU (using crypto.WithSender() option) over A256KW as
-// 		per https://tools.ietf.org/html/rfc7518#appendix-A.2
-//  - KDF: Concat KDF as per https://tools.ietf.org/html/rfc7518#section-4.6
+//  - Key Unwrapping: `ECDH-ES` (no options) or `ECDH-1PU` (using crypto.WithSender() option in wrapKeyOpts) over either
```

Comment:
Like I said above, I don't understand why we need to pass in options for the algorithms to use when all this information is in the recipient header.

Reply:
see my answer to the previous question above ..